app/services/cache/MongoCacheClient.scala

```diff
@@ -114,53 +114,58 @@ class MongoCacheClient(appConfig: AppConfig, db: () => DefaultDB, applicationCry
   /**
     * Inserts data into the cache with the specified key. If the data does not exist, it will be created.
     */
-  def createOrUpdate[T](id: String, data: T, key: String)(implicit writes: Writes[T]): Future[Cache] = {
+  def createOrUpdateWithCacheMiss[T](id: String, newId: String, data: T, key: String)(implicit writes: Writes[T]): Future[Cache] = {
     val jsonData = if (appConfig.mongoEncryptionEnabled) {
       val jsonEncryptor = new JsonEncryptor[T]()
       Json.toJson(Protected(data))(jsonEncryptor)
     } else {
       Json.toJson(data)
     }
 
-    fetchAll(id) flatMap { maybeCache =>
-      val cache = maybeCache.getOrElse(Cache(id, Map.empty))
+    fetchAll(newId, false) flatMap { maybeNewCache =>
```

Comment:
Could use a for comprehension, rather than a nesting

Reply:
like above